external/kafka-0-10-token-provider/src/test/scala/org/apache/spark/kafka010/KafkaTokenUtilSuite.scala

```diff
@@ -239,24 +239,17 @@ class KafkaTokenUtilSuite extends SparkFunSuite with KafkaDelegationTokenTest {
     assert(jaasParams.contains(tokenPassword1))
   }
 
-  test("needTokenUpdate without security credentials enabled should return false") {
```

Comment:
In order to help reviewers since `needTokenUpdate` behaviour changed this test doesn't make sense so removed. From now on `needTokenUpdate` can be called only when it's checked that the Kafka related delegation token provider is enabled. This happens in `KafkaDataConsumer`.

Reply:
Shall we add some comments on `needTokenUpdate` to explain when it should be called?